src/main/java/seedu/duke/command/ListAndStatsCommand.java

```diff
@@ -1,139 +1,145 @@
-                    + "as an invalid combination of tags has been given.");
-            throw new GlobalUnsupportedTagException();
-        } else if (containMonthYear() == TRUE_INVALID_OR) {
-            // Throws a missing tag if number and period was not given together
-            datedTransactionsLogger.log(Level.WARNING, "An exception has been caught as "
-                    + "a month was given without a year.");
-            throw new GlobalMissingYearTagException();
-        } else if (containPeriodNumber() == TRUE_OR) {
-            // Throws a missing tag if number and period was not given together
-            datedTransactionsLogger.log(Level.WARNING, "An exception has been caught as "
-                    + "number and period needs to be given together.");
-            throw new GlobalMissingPeriodNumberTagException();
-        }
-    }
-
-    /**
-     * Gets the complete transactions list by date intervals.
-     *
-     * @param transactions  An instance of the TransactionList class.
-     * @return An array list containing all transactions recorded in specified date interval.
-     */
-    public ArrayList<Transaction> getTimeTransactions(TransactionList transactions) {
-        ArrayList<Transaction> timeTransactions = transactions.getTransactions();
-
-        if (containMonthYear() == TRUE_AND) {
-            timeTransactions = transactions.getTransactionsByMonth(year, month);
-        } else if (containMonthYear() == TRUE_OR) {
-            assert month == UNDEFINED_PARAMETER;
-            timeTransactions = transactions.getTransactionsByYear(year);
-        } else if (containPeriodNumber() == TRUE_AND && period == MONTHS) {
-            timeTransactions = transactions.getTransactionsByMonthRange(LocalDate.now(), number);
-        } else if (containPeriodNumber() == TRUE_AND && period == WEEKS) {
-            timeTransactions = transactions.getTransactionsByWeekRange(LocalDate.now(), number);
-        }
-
-        return timeTransactions;
-    }
-}
+package seedu.duke.command;
+
+//@@author chydarren
+import seedu.duke.data.TransactionList;
+import seedu.duke.data.transaction.Transaction;
+import seedu.duke.exception.GlobalMissingPeriodNumberTagException;
+import seedu.duke.exception.GlobalMissingYearTagException;
+import seedu.duke.exception.GlobalUnsupportedTagException;
+import seedu.duke.exception.MoolahException;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Represents an object that can be inherited by List and Stats command objects.
+ */
+public abstract class ListAndStatsCommand extends Command {
+    //@@author chydarren
+    private static final int UNDEFINED_PARAMETER = -1;
+    private static final int TRUE_AND = 1;
+    private static final int TRUE_OR = 2;
+    private static final int TRUE_INVALID_OR = 3;
+    private static final int FALSE = 0;
```

Comment:
I feel that the naming for these is quite confusing, and it is quite difficult to comprehend what they represent.

Reply:
Thank you for this feedback, the variables have been renamed.